### PR TITLE
VZ-8248 Disabling resource requirements validation

### PR DESCRIPTION
**Pull request by AbeHern**
Added boolean param to disabled resource requirements warnings and subsequent logic to test and toggle when desired.

### CODE REVIEW

**Review by bobdonat — APPROVED** [review]

**Review by anders-swanson — APPROVED** [review]

**Review by anders-swanson — APPROVED** [review]

**Review by bobdonat — APPROVED** [review]

File: platform-operator/helm_config/charts/verrazzano-platform-operator/values.yaml [hunk 1]
```diff
@@ -39,3 +39,5 @@ webhookAffinity:
                   - verrazzano-platform-operator-webhook
           topologyKey: kubernetes.io/hostname
         weight: 100
+webhooks:
+  resourceValidation: false
```
Comment (kminder) [inline platform-operator/helm_config/charts/verrazzano-platform-operator/values.yaml:43]: Doesn't this need to be templated?

File: pkg/k8sutil/apply_yaml_test.go [hunk 2]
```diff
@@ -141,9 +141,8 @@ func TestApplyFNonSpec(t *testing.T) {
 	assert.Equal(t, 1, len(saUpdated.ImagePullSecrets))
 	assert.Equal(t, "verrazzano-container-registry", saUpdated.ImagePullSecrets[0].Name)
 
-	assert.NotEmpty(t, saUpdated.Secrets)
-	assert.Equal(t, 1, len(saUpdated.Secrets))
-	assert.Equal(t, "verrazzano-platform-operator-token", saUpdated.Secrets[0].Name)
```
Comment (kminder) [inline pkg/k8sutil/apply_yaml_test.go:146]: Why was this removed?

File: pkg/k8sutil/apply_yaml.go [hunk 3]
```diff
@@ -209,6 +209,25 @@ func (y *YAMLApplier) applyAction(obj *unstructured.Unstructured) error {
 				return err
 			}
 		}
+		// Delete any keys in server obj not included in the client fields.
```
Comment (kminder) [inline pkg/k8sutil/apply_yaml.go:212]: Why was this added?

File: platform-operator/helm_config/charts/verrazzano-platform-operator/values.yaml (hunk 1 above)
Comment (AbeHern) [reply platform-operator/helm_config/charts/verrazzano-platform-operator/values.yaml:43]: Here is where the value is defaulted. Not sure if it needs to be formatted here but it is formatted in validatingwebhookconfiguration.yaml

File: pkg/k8sutil/apply_yaml_test.go (hunk 2 above)
Comment (AbeHern) [reply pkg/k8sutil/apply_yaml_test.go:146]: The file used in this test has no Secrets, so the test kept breaking as it was expecting to have a secret. So it was removed

File: pkg/k8sutil/apply_yaml.go (hunk 3 above)
Comment (AbeHern) [reply pkg/k8sutil/apply_yaml.go:212]: I had Bob help me with this. From what I've understood, last time I merged and broke master it was failing at Upgrade Path. There was an issue with the vz cli that was still expecting there to be a validating webhook when the new operator.yaml does not indeed have one. So This pruning essentially takes care of this.

File: platform-operator/helm_config/charts/verrazzano-platform-operator/values.yaml (hunk 1 above)
Comment (kminder) [reply platform-operator/helm_config/charts/verrazzano-platform-operator/values.yaml:43]: My bad.  I was thinking this was a template.

File: pkg/k8sutil/apply_yaml_test.go (hunk 2 above)
Comment (kminder) [reply pkg/k8sutil/apply_yaml_test.go:146]: ok

File: pkg/k8sutil/apply_yaml.go (hunk 3 above)
Comment (kminder) [reply pkg/k8sutil/apply_yaml.go:212]: I see.  Thanks.
